mula/scheduler/connectors/services/katalogus.py

```diff
@@ -22,69 +22,71 @@ def __init__(
 
-        self.lock = threading.Lock()
-
         # For every organisation we cache its plugins, it references the
         # plugin-id as key and the plugin as value.
-        self.organisations_plugin_cache: dict_utils.ExpiringDict = dict_utils.ExpiringDict(lifetime=cache_ttl)
+        self.plugin_cache_lock = threading.Lock()
+        self.plugin_cache: dict_utils.ExpiringDict = dict_utils.ExpiringDict(lifetime=cache_ttl)
 
         # For every organisation we cache on which type of object (consumes)
         # the boefjes consume, it references the object type (consumes)
         # as the key and a dict of boefjes as value.
-        self.organisations_boefje_type_cache: dict_utils.ExpiringDict = dict_utils.ExpiringDict(lifetime=cache_ttl)
+        self.boefje_cache_lock = threading.Lock()
+        self.boefje_cache: dict_utils.ExpiringDict = dict_utils.ExpiringDict(lifetime=cache_ttl)
 
         # For every organisation we cache on which type of object (consumes)
         # the normalizers consume, it references the object type (consumes)
         # as the key and a dict of normalizers as value.
-        self.organisations_normalizer_type_cache: dict_utils.ExpiringDict = dict_utils.ExpiringDict(lifetime=cache_ttl)
+        self.normalizer_cache_lock = threading.Lock()
+        self.normalizer_cache: dict_utils.ExpiringDict = dict_utils.ExpiringDict(lifetime=cache_ttl)
 
         # For every organisation we cache which new boefjes for an organisation
         # have been enabled.
-        self.organisations_new_boefjes_cache: dict = {}
+        self.new_boefjes_cache_lock = threading.Lock()
+        self.new_boefjes_cache: dict = {}
 
         # Initialise the cache.
         self.flush_caches()
 
     def flush_caches(self) -> None:
-        self.flush_organisations_plugin_cache()
-        self.flush_organisations_normalizer_type_cache()
-        self.flush_organisations_boefje_type_cache()
+        self.flush_plugin_cache()
+        self.flush_normalizer_cache()
+        self.flush_boefje_cache()
 
-    def flush_organisations_plugin_cache(self) -> None:
+    def flush_plugin_cache(self) -> None:
         self.logger.debug("Flushing the katalogus plugin cache for organisations")
 
-        with self.lock:
+        with self.plugin_cache_lock:
             # First, we reset the cache, to make sure we won't get any ExpiredError
-            self.organisations_plugin_cache.expiration_enabled = False
-            self.organisations_plugin_cache.reset()
+            self.plugin_cache.expiration_enabled = False
+            self.plugin_cache.reset()
 
             orgs = self.get_organisations()
             for org in orgs:
-                if org.id not in self.organisations_plugin_cache:
-                    self.organisations_plugin_cache[org.id] = {}
-                    self.organisations_new_boefjes_cache[org.id] = {}
+                if org.id not in self.plugin_cache:
+                    self.plugin_cache[org.id] = {}
```

Comment:
You can use `dict.setdefault` for this purpose

```suggestion
                plugin_cache.setdefault(org.id, {})
```

Reply:
Unfortunately `self.plugin_cache` is of type `ExpiringDict` that doesn't implement the `setdefault()` method. I've added the method to the `ExpiringDict` to make this possible as well.

Reply:
Cool. Maybe a bit late but something to think about (for another PR); if `ExpiringDict` is supposed to act like a dict, maybe it could extend a `dict` or `UserDict` instead